packages/dev/core/src/Meshes/Builders/shapeBuilder.ts

```diff
@@ -230,13 +250,56 @@ function _ExtrudeShapeGeneric(
         scaleFunction: Nullable<{ (i: number, distance: number): number }>,
         rotateFunction: Nullable<{ (i: number, distance: number): number }>,
         cap: number,
-        custom: boolean
+        custom: boolean,
+        adjustFrame: boolean
     ) => {
         const tangents = path3D.getTangents();
         const normals = path3D.getNormals();
         const binormals = path3D.getBinormals();
         const distances = path3D.getDistances();
-
+        if (adjustFrame) {
+            /* fix tangents,normals, binormals */
+            for (let i = 0; i < tangents.length; i++) {
+                if (tangents[i].x == 0 && tangents[i].y == 0 && tangents[i].z == 0) {
+                    tangents[i].x = tangents[i - 1].x;

```

Comment:
if this is a vector3 (which I am pretty sure it is?), why not use `.copyFrom()` instead? it's just cleaner, as copyFrom does exactly what these lines do